### DISCUSSION TITLE
Automation identifying window title in one windows system but not antoher

### DISCUSSION THREAD

**Discussion by karges612** [pywinauto/pywinauto#1279] [Q&A]
Belo is my code , which is perfectly working fine in my system , but when i deploy the code as python exe in another system. it still manages to trigger the window. but not detecting the window , in other words  its not typing anything in the window. 
and the error says that there is no such entry as login in the dictionary.

`MatchError("Could not find 'Login' in 'dict_keys(['Dialog', 'maifestMain - Shortcut', 'Program Manager', 'Program ManagerPane',`

```
app = Application(backend = "uia").start(settings.apppath)
LoginWindow = Desktop(backend="uia").Login
            try:
                LoginWindow.UsernameEdit.type_keys("rputra1",pause=.1)
            except:
                pass
            try:
                LoginWindow.Edit2.type_keys("Welcome123",pause=.1)
            except:
                pass
            LoginWindow.LOGINButton.click()
```

**Comment by vasily-v-ryabov** [comment]
Maybe "Login" window is a child of other top level window? It's pretty typical situation for "uia" backend. Don't know why it's a top level window on one system. Is there a big difference in OS versions? I mean something like Win10 vs WinServer2022 or whatever.

**Reply by vasily-v-ryabov** [reply]
For example, you can walk for all top windows and dump all elements with `control_type="Window"`:
```python
for w in Desktop(backend="uia").windows():
    print(f'Top window "w.window_text()"')
    print("    " + str([elem.window_text() for elem in w.children(control_type="Window")]))
```
or just `w.children()` without `control_type` at all.

**Reply by karges612** [reply]
@vasily-v-ryabov including a time out property in the start method actually solved the problem.
how did i know that a time out might solved the issue ? 
in one of the 7 attempts to run the automation the bot detected the window and typed the user name and password. but for 95 % of the time it never detected. that is when i realized.
` Application(backend = "uia").start(settings.apppath, timeout=10)`
but to answer your question . yes i developed in windows 11 and deployed in windows 10. and i have a very interesting case study on this automation.  
while automation runs fine in my system. in the windows 10 system its encountering some ui elements which is not even visible otherwise . its  a JFX windows application.
i want to enlighten the group by giving more information on this. is there any discord or meetup planned ? 
i can document everything and post here also.

**Reply by vasily-v-ryabov** [reply]
Almost every intermittent problem is a timing problem, it's pretty typical for GUI automation. I think Win10/11 difference report should be in a JavaFX community because it depends on application side support which controls are visible. We have only Gitter chat which is not very active.

**Reply by karges612** [reply]
@vasily-v-ryabov  i have an interesting scenario. its not even the OS. the code is working fine and clicking right buttons in remote desktop mode (of windows 10 system). but when some one actually logins to the same windows 10 system with the same account
(which i used for RDC) the code is not working. as in , the control identifiers prints some elements which are not part of the current window . but which might be part of same application but not visible. basically controls related to some other roles in the application. 
recap : 
App : javaFX application 
OS : windows 10. 
essentially this is not a problem while shifting the code from 11 to 10 . but this happens in 10 itself in actual login but not remote login!! am taking screen share of the remote system and doing the development!
